app/Auth0/Auth0ClientsForIntegration.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Auth0;
+
+use InvalidArgumentException;
+use Ramsey\Uuid\UuidInterface;
+
+final class Auth0ClientsForIntegration
+{
+    private array $auth0ClientPerTenant = [];
```

Comment:
I presume you can't type this array more specifically in php? 

Something like:

```php
private [ string => Auth0Client ] $auth0ClientPerTenant = [];
```

Reply:
Or that is only possible with comment syntax like below?:

```php
 /**
     * @var Auth0TenantSDK[]
     */
```

Reply:
In PHPDoc it would have to be:

```
@var Auth0Client[]
```

(https://docs.phpdoc.org/latest/guide/references/phpdoc/types.html#arrays)

However then the type of the array keys is missing (since this is an associative array, meaning that the keys are strings instead of integers)

In Psalm it can be represented as:
```
@var array<string, Auth0Client>
```
https://psalm.dev/docs/annotating_code/type_syntax/array_types/

But I'm not 100% sure that PHPStan and other tools always work with that. Honestly I'm not sure it's worth the effort for such simple cases because it's not a native language construct, so the only thing we're adding is some kind of safety _depending on the tools we use_. Also because when PHP adds native support for this, it will probably not be by using `@` annotations.

Reply:
But this is not generics I think? The value type is always the same (`Auth0Client`) regardless of the instance of `Auth0ClientsForIntegration`

Reply:
And again it's only for support in specific tools like PHPStan, which is not complaining in this case

Reply:
That is because it is disabled here https://github.com/cultuurnet/publiq-platform/blob/main/phpstan.neon#L16